### PR TITLE
Fix compilation issues on gcc 8

### PR DESCRIPTION
**Pull request by 8tab**

### CODE REVIEW

**Comment by adaxi123** [comment]
@8tab Hi，you've got failed in travis ci. Please modify your code according to the error info.

**Comment by 8tab** [comment]
Hi, I see one error:
'paddle/fluid/pybind/inference_api.cc:288:  Extra space before ( in function call  [whitespace/parens] [4]'
But I did not modify that line. 
Is it the error I should focus on, or maybe I missed anything else?

**Comment by luotao1** [comment]
1. Please have a simple or detail description, now it's null.
2. Please use `pre-commit` to solve the travis-ci error: https://www.paddlepaddle.org.cn/documentation/docs/zh/1.5/advanced_usage/development/contribute_to_paddle/local_dev_guide.html#permalink-6--pre-commit-
3. > But I did not modify that line.

I think line 288 is combined with line 289, thus, you have a `Extra space`

File: cmake/flags.cmake [hunk 1]
```diff
@@ -132,6 +132,7 @@ set(COMMON_FLAGS
     -Wno-error=int-in-bool-context # Warning in Eigen gcc 7.2
     -Wimplicit-fallthrough=0 # Warning in tinyformat.h
     -Wno-error=maybe-uninitialized # Warning in boost gcc 7.2
+    -Wno-error=parentheses # Warning in boost gcc 8.2
```
Comment (luotao1) [inline cmake/flags.cmake:135]: How about move line134 and line135 to `boost.cmake`? You can finish it in next PR.

**Pull request closed without merging**